pkg/query/v2/logical/common.go

```diff
@@ -19,41 +19,32 @@ package logical
 
 import (
 	"bytes"
-	"time"
-
-	"github.com/pkg/errors"
-	"google.golang.org/protobuf/types/known/timestamppb"
 
 	modelv2 "github.com/apache/skywalking-banyandb/api/proto/banyandb/model/v2"
-	streamv2 "github.com/apache/skywalking-banyandb/api/proto/banyandb/stream/v2"
 	"github.com/apache/skywalking-banyandb/banyand/tsdb"
 	"github.com/apache/skywalking-banyandb/pkg/query/v2/executor"
 )
 
 type (
 	seekerBuilder func(builder tsdb.SeekerBuilder)
-	comparator    func(a, b *streamv2.Element) bool
+	comparator    func(a, b tsdb.Item) bool
 )
 
-func createTimestampComparator(sort modelv2.QueryOrder_Sort) comparator {
-	if sort == modelv2.QueryOrder_SORT_ASC {
-		return func(a, b *streamv2.Element) bool {
-			return a.GetTimestamp().AsTime().Before(b.GetTimestamp().AsTime())
+func createTimestampComparator(sortDirection modelv2.QueryOrder_Sort) comparator {
+	if sortDirection == modelv2.QueryOrder_SORT_ASC {
+		return func(a, b tsdb.Item) bool {
+			return a.Time() < b.Time()
 		}
 	}
 
-	return func(a, b *streamv2.Element) bool {
-		return a.GetTimestamp().AsTime().After(b.GetTimestamp().AsTime())
+	return func(a, b tsdb.Item) bool {
+		return a.Time() > b.Time()
 	}
 }
 
-func createMultiTagsComparator(fieldRefs []*FieldRef, sortDirection modelv2.QueryOrder_Sort) comparator {
-	tagFamilyIdx, tagIdx := fieldRefs[0].Spec.TagFamilyIdx, fieldRefs[0].Spec.TagIdx
-	return func(a, b *streamv2.Element) bool {
-		iTag, jTag := a.GetTagFamilies()[tagFamilyIdx].GetTags()[tagIdx], b.GetTagFamilies()[tagFamilyIdx].GetTags()[tagIdx]
-		lField, _ := getRawTagValue(iTag)
-		rField, _ := getRawTagValue(jTag)
-		comp := bytes.Compare(lField, rField)
+func createComparator(sortDirection modelv2.QueryOrder_Sort) comparator {
+	return func(a, b tsdb.Item) bool {
+		comp := bytes.Compare(a.SortedField(), b.SortedField())
 		if sortDirection == modelv2.QueryOrder_SORT_ASC {
```

Comment:
tsdb defaults sort order to ASC, which means Unspecified order should be ASC. please follow this way here

Reply:
I am skeptical about this default behavior.

Reply:
Fixed